src/AdvanceSteelNodes/base/Utils.cs

```diff
@@ -562,6 +564,44 @@ public static IEnumerable<SteelDbObject> GetDynObjects(IEnumerable<string> handl
       return retListOfSteelObjects;
     }
 
+    public static IEnumerable<SteelDbObject> GetDynObjects(int objectSelectionType)
+    {
+      var retListOfSteelObjects = new List<SteelDbObject>();
+      using (var ctx = new DocContext())
+      {
+        List<string> ret = new List<string>() { };
+
+        ObjectId[] OIDxs = null;
+
+        ClassTypeFilter xFilter = new ClassTypeFilter();
+        xFilter.RejectAllFirst();
+        xFilter.AppendAcceptedClass((FilerObject.eObjectType)objectSelectionType);
+
+        DatabaseManager.GetModelObjectIds(out OIDxs, xFilter);
+        List<ObjectId> OIDx = OIDxs.ToList<ObjectId>();
+        if (OIDx.Count > 0)
+        {
+          for (int i = 0; i < OIDx.Count; i++)
+          {
+            FilerObject obj = FilerObject.GetFilerObject(OIDx[i]);
+            if (obj != null)
+            {
+              string objHandle = obj.Handle;
+              if (avaliableSteelObjects.ContainsKey(obj.Type()))
+              {
+                SteelDbObject foundSteelObj = avaliableSteelObjects[obj.Type()](objHandle);
+                foundSteelObj.IsOwnedByDynamo = false;
+                foundSteelObj.Handle = objHandle;
+                retListOfSteelObjects.Add(foundSteelObj);
+              }
+            }
+            obj.Dispose();

```

Comment:
not sure if you should call the Dispose here. But, If you must call it, you should use the "using" keyword.
using(FilerObject obj = FilerObject.GetFilerObject(OIDx[i])) 
{
...
}
and remove the line that calls Dispose